examples/progress-dashboard/app.py

```diff
@@ -0,0 +1,205 @@
+
+import argilla as rg
+import gradio as gr
+import pandas as pd
+import plotly.colors as colors
+import plotly.graph_objects as go
+
+client = rg.Argilla(
+    api_url=os.getenv("ARGILLA_API_URL"),
+    api_key=os.getenv("ARGILLA_API_KEY"),
+)
+
+
+def get_progress(dataset: rg.Dataset) -> dict:
+    dataset_progress = dataset.progress(with_users_distribution=True)
+
+    total, completed = dataset_progress["total"], dataset_progress["completed"]
+    progress = (completed / total) * 100 if total > 0 else 0
+
+    return {
+        "total": total,
+        "annotated": completed,
+        "progress": progress,
+        "users": {
+            username: user_progress["completed"].get("submitted")
+            for username, user_progress in dataset_progress["users"].items()
+        },
+    }
+
+
+def create_gauge_chart(progress):
+    fig = go.Figure(
+        go.Indicator(
+            mode="gauge+number+delta",
+            value=progress["progress"],
+            title={"text": "Dataset Annotation Progress", "font": {"size": 24}},
+            delta={"reference": 100, "increasing": {"color": "RebeccaPurple"}},
+            number={"font": {"size": 40}, "valueformat": ".1f", "suffix": "%"},
+            gauge={
+                "axis": {"range": [None, 100], "tickwidth": 1, "tickcolor": "darkblue"},
+                "bar": {"color": "deepskyblue"},
+                "bgcolor": "white",
+                "borderwidth": 2,
+                "bordercolor": "gray",
+                "steps": [
+                    {"range": [0, progress["progress"]], "color": "royalblue"},
+                    {"range": [progress["progress"], 100], "color": "lightgray"},
+                ],
+                "threshold": {
+                    "line": {"color": "red", "width": 4},
+                    "thickness": 0.75,
+                    "value": 100,
+                },
+            },
+        )
+    )
+
+    fig.update_layout(
+        annotations=[
+            dict(
+                text=(
+                    f"Total records: {progress['total']}<br>"
+                    f"Annotated: {progress['annotated']} ({progress['progress']:.1f}%)<br>"
+                    f"Remaining: {progress['total'] - progress['annotated']} ({100 - progress['progress']:.1f}%)"
+                ),
+                # x=0.5,
+                # y=-0.2,
+                showarrow=False,
+                xref="paper",
+                yref="paper",
+                font=dict(size=16),
+            )
+        ],
+    )
+
+    fig.add_annotation(
+        text=(
+            f"Current Progress: {progress['progress']:.1f}% complete<br>"
+            f"({progress['annotated']} out of {progress['total']} records annotated)"
+        ),
+        xref="paper",
+        yref="paper",
+        x=0.5,
+        y=1.1,
+        showarrow=False,
+        font=dict(size=18),
+        align="center",
+    )
+
+    return fig
+
+
+def create_treemap(user_annotations, total_records):
```

Comment:
@davidberenstein1957 Do you think we could simplify this?

Reply:
I think we can maybe just add percentages to the table? I am not 100% sure if the treemap adds a lot of value compared to that.